### ISSUE TITLE
export template command fails in 7.0

### ISSUE THREAD

**Issue by andrewkroh** [elastic/beats#10098]
The `export template` command fails with a cryptic error message and stack when using a customized `setup.template.pattern` value. This probably stems from this code not being updated to have `agent` instead of `beat`.

```
 ./filebeat export template -E setup.template.pattern='filebeat-%{[agent.version]}-*'
Error generating template: key not found
github.com/elastic/beats/libbeat/common.init
	/Users/akroh/go/src/github.com/elastic/beats/libbeat/common/mapstr.go:40
github.com/elastic/beats/libbeat/beat.init
	<autogenerated>:1
github.com/elastic/beats/filebeat/cmd.init
	<autogenerated>:1
main.init
	<autogenerated>:1
runtime.main
	/Users/akroh/.gvm/versions/go1.11.4.darwin.amd64/src/runtime/proc.go:189
runtime.goexit
	/Users/akroh/.gvm/versions/go1.11.4.darwin.amd64/src/runtime/asm_amd64.s:1333
```

This code still uses `beat` rather than `agent`.

https://github.com/elastic/beats/blob/6321c5da066ba37673d0e8b112ca594f67745709/libbeat/template/template.go#L70-L78

Additionally the error message should be improved and the stack not included (don't use `%+v` to log the error to stdout). The error message would be more help if it included `agent.version` (which I assume is the missing key).

- Version: filebeat version 7.0.0 (amd64), libbeat 7.0.0 [2f9c1fe2d2de0827e528b3ba1b333f93bdf31258 built 2019-01-16 02:55:33 +0000 UTC]

**Comment by andrewkroh** [comment]
This affects automatic template loading too.

```
{"level":"error","timestamp":"2019-01-16T03:33:47.007Z","caller":"pipeline/output.go:100","message":"Failed to connect to backoff(elasticsearch(https://4c9150723e40c6b7af6b8e651ccf8a35.us-east-1.aws.found.io:9243)): Connection marked as failed because the onConnect callback failed: Error loading Elasticsearch template: error creating template instance: key not found"}
```

**Comment by andrewkroh** [comment]
Fixed via #10078.
